src/hooks/useCalculatedSize.tsx

```diff
@@ -0,0 +1,9 @@
+import { useMediaQuery } from '@mui/material';
+import { componentSize, mediaQueries } from '../components/DesignSystem/DesignSystem';
+
+const useCalculatedSize = (component: keyof typeof componentSize) => {
```

Comment:
I'm not a big fan of the JS media queries as they might result in performance hit. Can't we solve this using CSS?